src/sentry/api/serializers/models/team.py

```diff
@@ -187,20 +197,43 @@ def serialize(
         return d
 
 
+def get_scim_teams_members(
+    team_list: Sequence[Team],
+) -> MutableMapping[Team, MutableSequence[MutableMapping[str, Any]]]:
+    members = RangeQuerySetWrapper(
+        OrganizationMember.objects.filter(teams__in=team_list).select_related("user"),
```

Comment:
This query will end up returning duplicate `OrganizationMember` rows for members that belong to many teams, since it does a join:
```
SELECT "sentry_organizationmember"."id", "sentry_organizationmember"."organization_id", "sentry_organizationmember"."user_id", "sentry_organizationmember"."email", "sentry_organizationmember"."role", "sentry_organizationmember"."flags", "sentry_organizationmember"."token", "sentry_organizationmember"."date_added", "sentry_organizationmember"."token_expires_at", "sentry_organizationmember"."has_global_access", "sentry_organizationmember"."inviter_id", "sentry_organizationmember"."invite_status", "sentry_organizationmember"."type" 
FROM "sentry_organizationmember" 
INNER JOIN "sentry_organizationmember_teams" ON ("sentry_organizationmember"."id" = "sentry_organizationmember_teams"."organizationmember_id") WHERE "sentry_organizationmember_teams"."team_id" IN (1, 2, 3)
```

You can use `.distinct("id")` to de-dupe it.

Reply:
great catch, addressed. 👍🏼 I suppose the lesson is that when you filter on a FK column, it will do a join instead of making a subquery. will be more mindful looking at sql generated by these querysets going forward

Reply:
Yeah, you can explicitly make a subquery if you prefer as well. Something like
`OrganizationMember.objects.filter(id__in=OrganizationMemberTeam.objects.filter(team__in=team_list).values_list("organizationmember_id", flat=True))`